controllers/create_pvc.go

```diff
@@ -0,0 +1,85 @@
+// Copyright Contributors to the Open Cluster Management project
+package controllers
+
+import (
+	"context"
+
+	searchv1alpha1 "github.com/stolostron/search-v2-operator/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (r *SearchReconciler) configurePVC(ctx context.Context, instance *searchv1alpha1.Search) bool {
+	if err := r.createPVC(ctx, instance); err != nil {
+		return false
+	}
+	return true
+}
+
+func (r *SearchReconciler) createPVC(ctx context.Context, instance *searchv1alpha1.Search) error {
```

Comment:
What would happen if the PVC exists with a different size? Should this logic be updated to create or update PVC?

Reply:
Per discussion with @jlpadilla @bjoydeep we are not planning to support PVC resize.

Reply:
I'm OK if we don't support PVC resize, but how do we prevent users from changing the size value after the CR is created?

Reply:
Users can change it , but it will not have any impact . If Users really want to change size, they have to bring down search and Update the PVC size and restart search .

Reply:
ok, I understand the challenge and I don't want to block this PR, so let's merge but keep this discussion open.

For the user it will be frustrating that their changes don't take effect. I'd like to see an error state or something that indicates that we are unable to apply the requested desired state.